packages/web/src/hooks/useVerifyConnection.js

```diff
@@ -0,0 +1,26 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import api from 'helpers/api';
+
+export default function useVerifyConnection() {
+  const queryClient = useQueryClient();
+
+  const query = useMutation({
+    mutationFn: async (connectionId) => {
+      try {
+        const { data } = await api.post(
+          `/v1/connections/${connectionId}/verify`,
+        );
+
+        return data;
+      } catch {
+        throw new Error('Failed while verifying connection!');
+      }
+    },
+    onSuccess: (data) => {
+      const appKey = data?.data.key;
+      queryClient.invalidateQueries(['apps', appKey, 'connections']);
```

Comment:
I think we need to iterate the function signature here, like `{ queryKey: ['apps', appKey, 'connections'] }`

Reply:
True, I will fix that.